$OpenBSD$

OpenBSD cp command does not have "-a" option, use "-pPR" instead.
sed "i" command expects \ followed by text...

Index: Makefile.checker
--- Makefile.checker.orig
+++ Makefile.checker
@@ -36,17 +36,53 @@ CHECKERDEPS := $(addsuffix .d, $(CHECKMLDFILE) $(CHECK
 
 # Copied files
 checker/esubst.mli: kernel/esubst.mli
-	cp -a $< $@
-	sed -i.bak '1i(* AUTOGENERATED FILE: DO NOT EDIT *)\n\n\n\n\n\n\n\n' $@ && rm $@.bak
+	cp -pPR $< $@
+	sed -i.bak -E '1i\
+(* AUTOGENERATED FILE: DO NOT EDIT *)\
+\
+\
+\
+\
+\
+\
+\
+' $@ && rm $@.bak
 checker/esubst.ml: kernel/esubst.ml
-	cp -a $< $@
-	sed -i.bak '1i(* AUTOGENERATED FILE: DO NOT EDIT *)\n\n\n\n\n\n\n\n' $@ && rm $@.bak
+	cp -pPR $< $@
+	sed -i.bak -E '1i\
+(* AUTOGENERATED FILE: DO NOT EDIT *)\
+\
+\
+\
+\
+\
+\
+\
+' $@ && rm $@.bak
 checker/names.mli: kernel/names.mli
-	cp -a $< $@
-	sed -i.bak '1i(* AUTOGENERATED FILE: DO NOT EDIT *)\n\n\n\n\n\n\n\n' $@ && rm $@.bak
+	cp -pPR $< $@
+	sed -i.bak -E '1i\
+(* AUTOGENERATED FILE: DO NOT EDIT *)\
+\
+\
+\
+\
+\
+\
+\
+' $@ && rm $@.bak
 checker/names.ml: kernel/names.ml
-	cp -a $< $@
-	sed -i.bak '1i(* AUTOGENERATED FILE: DO NOT EDIT *)\n\n\n\n\n\n\n\n' $@ && rm $@.bak
+	cp -pPR $< $@
+	sed -i.bak -E '1i\
+(* AUTOGENERATED FILE: DO NOT EDIT *)\
+\
+\
+\
+\
+\
+\
+\
+' $@ && rm $@.bak
 
 ifeq ($(BEST),opt)
 $(CHICKEN): checker/check.cmxa checker/main.mli checker/main.ml
